majordodo-core/src/main/java/majordodo/task/Task.java

```diff
@@ -234,10 +247,42 @@ Task cloneForSnapshot() {
         copy.type = this.type;
         copy.attempts = this.attempts;
         copy.maxattempts = this.maxattempts;
+        copy.requestedStartTime = this.requestedStartTime;
         copy.executionDeadline = this.executionDeadline;
         copy.slot = this.slot;
         copy.resources = this.resources;
         return copy;
     }
 
+    @Override
+    public long getDelay(TimeUnit unit) {
+        long delay = requestedStartTime - System.currentTimeMillis();
+        if (delay < 0) {
+            delay = 0;
+        }
+        switch (unit) {

```

Comment:
are you sure ?
I think usually this kind of code does not enumerate all possible cases. IMHO the is a bug here

this is a better code:
`return unit.convert(requestedStartTime - System.currentTimeMillis(), TimeUnit.MILLISECONDS);`